src/osp/Universe.h

```diff
@@ -173,6 +183,14 @@ TYPESAT_T& Universe::type_register(ARGS_T&& ... args)
     return toReturn;
 }
 
+template<typename SATTYPE_T>
+SATTYPE_T& Universe::sat_type_find()
+{
+    MapSatType::iterator it = sat_type_find(SATTYPE_T::smc_name);
+    assert(it != m_satTypes.end());
+    return static_cast<SATTYPE_T&>(*it->second);
+}
+
```

Comment:
struct UCompTransformTraj
{
    // Relative to m_parent
    Vector3s m_position;
    // In 'global' space
    Quaternion m_rotation;
    Satellite m_parent; // Set only by trajectory
    ISystemTrajectory *m_trajectory{nullptr}; // get rid of this some day
    unsigned m_index; // index in trajectory's vector
    // set this to true when you modify something the trajectory might use
    bool m_dirty{true};
    // move this somewhere else eventually
    std::string m_name;
};


Probably use size_t instead of "unsigned".

If you don't want size_t, then better would be e.g. std::uint64_t or something similar.

Reply:
Additionally, consider struct-packing.

you'll always get the least amount of padding added by the compiler if you order your member variables with largest alignas() to smallest alignas(), meaning your m_dirty bool would go last, and std::string very well may go first.